src/main/java/ch/njol/skript/expressions/ExprVectorFromDirection.java

```diff
@@ -0,0 +1,99 @@
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Direction;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.event.Event;
+import org.bukkit.util.Vector;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Vectors - Create from Direction")
+@Description({
+	"Creates vectors from given directions.",
+	"Relative directions are relative to the origin, (0, 0, 0). Therefore, the vector from the direction 'forwards' is (0, 0, 1)."
+})
+@Examples({
+	"set {_v} to vector from direction upwards",
+	"set {_v} to vector in direction of player",
+	"set {_v} to vector in horizontal direction of player",
+	"set {_v} to vector from facing of player",
+	"set {_v::*} to vectors from north, south, east, and west"
+})
+@Since("INSERT VERSION")
+public class ExprVectorFromDirection extends SimpleExpression<Vector> {
+
+	static {
+		Skript.registerExpression(ExprVectorFromDirection.class, Vector.class, ExpressionType.SIMPLE,
+				"vector[s] [from|in] [direction] %directions%",
+				"%directions% vector[s]");
+	}
+
+	private Expression<Direction> direction;
+	private static final Location DEFAULT_LOCATION = new Location(null, 0, 0, 0);
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		direction = (Expression<Direction>) exprs[0];
+		if (matchedPattern == 1) {
+			if (!(direction instanceof ExprDirection)) {
+				Skript.error("The direction in '%directions% vector[s]' can not be a variable. Use the direction expression instead: 'northwards vector'.");
+				return false;
+			}
+		}
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected Vector[] get(Event event) {
+		Direction[] directions = this.direction.getArray(event);
+		Vector[] vectors = new Vector[directions.length];
+		for (int i = 0; i < directions.length; i++) {
+			vectors[i] = directions[i].getDirection(DEFAULT_LOCATION); // all relative directions are relative to the default location
+		}
+		return vectors;
```

Comment:
```suggestion
		return direction.stream(event);
				.map(direction -> direction.getDirection(DEFAULT_LOCATION))
				.toArray(Vector[]::new);
```
If you make a method called Direction#toVector() (which is essentially what you're rebuilding assuming it's not relative) you can use Direction::toVector as the map.

Reply:
Good idea!